### PR TITLE
docs(Observable): add documentation for Observable and operators

### LINKED ISSUE #5642: docs(Observable): document how to compose operators
After upgrading to alpha.48 I noticed that the http module no longer can stand on its own...
According to the examples we now have to import ```rxjs/operators/map``` to even get access to the essential map() method.

I am wondering, isn't this a bit clunky and unnecessary? Why reveal the dependency on rx.js like this?
Shouldn't this be abstracted from us? Seems like we also have to include rxjs separately in addition to Angular 2??

If rx is this visible in Angular 2.0, why provide map() etc via the http module at all? Can't you just use rx.js directly?

P.S. This is not listed as a breaking change in the change log :-)

### PR DESCRIPTION
**Pull request by jeffbcross**
Closes #5642

### CODE REVIEW

File: modules/angular2/src/facade/async.ts [hunk 1]
```diff
@@ -164,6 +164,94 @@ export class EventEmitter<T> extends Subject<T> {
...
+ * ```
+ *
+ * RxJS provides many "combinators" to extend the value of `Observable`. Combinators
+ * are similar to `Array` functions like `map`, `reduce`, `filter`. They are applied
+ * to each value emitted by an `Observable`. Some combinators can also support waiting
+ * for asynchronous values before continuing.
+
+ * The `Observable` in Angular currently doesn't provide any combinators by default.
+ * So it's necessary to explicitly import any combinators that an application requires.
+ * There are two ways to import RxJS combinators: pure and impure. The "pure" approach
```
Comment (robwormald) [inline modules/angular2/src/facade/async.ts]: perhaps? mention that there are three - pure, impure, and `import {Observable} from 'rxjs'` which gives you the kitchen sink.

Comment (jeffbcross) [inline modules/angular2/src/facade/async.ts]: :+1:

File: modules/angular2/src/facade/async.ts [hunk 2]
```diff
@@ -164,6 +164,94 @@ export class EventEmitter<T> extends Subject<T> {
...
+ *
+ * `Observable` is also being considered for inclusion in the
+ * [ECMAScript spec](https://github.com/zenparsing/es-observable).
+ *
+ * ## Example
+ *
+ * A simple example of using an `Observable` is a timer `Observable`.
+ *
+ * ```typescript
+ * import {Observable} from 'rxjs/Observable';
```
Comment (alxhub) [inline modules/angular2/src/facade/async.ts]: Extract example to a separate file using {@example} syntax.

File: modules/angular2/src/facade/async.ts [hunk 3]
```diff
@@ -164,6 +164,94 @@ export class EventEmitter<T> extends Subject<T> {
...
+ * var obs = new Observable(obs => {
+ *   var i = 0;
+ *   setInterval(_ => obs.next(++i), 1000);
+ * });
+ * obs
+ *   .map(i => `${i} seconds elapsed`)
+ *   .subscribe(msg => console.log(msg));
+ * ```
+ *
+ * Notice that the sequence of operations is now able to be expressed left-to-right
```
Comment (alxhub) [inline modules/angular2/src/facade/async.ts]: Maybe add a bit more guidance - users should be able to read the documentation for `Observable` and come away with a clear idea of how they should use it.

Comment (robwormald) [inline modules/angular2/src/facade/async.ts]: Also interval is maybe a bad example as there is a built in for it. Dunno.

**Comment by jeffbcross** [comment]
@alxhub and @robwormald I made some revisions and incorporated feedback. Mind giving it another read and see how well it reads?

**Comment by jeffbcross** [comment]
I fixed the clang failure (FWIW Travis was green other than clang)

File: modules/angular2/examples/facade/ts/async/observable.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// #docregion Observable
```
Comment (alxhub) [inline modules/angular2/examples/facade/ts/async/observable.ts]: Probably good to have // #enddocregion at the end.

**Comment by robwormald** [comment]
updates look good :+1:

**Comment by alexeagle** [comment]
Wrong alex?

**Comment by jelbourn** [comment]
@alexeagle Jeff just assigned it to me because he wanted a merge

**Pull request closed without merging**